homeassistant/components/camera/__init__.py

```diff
@@ -991,31 +986,17 @@ async def async_handle_play_stream_service(
     ]
     other_entity_ids = list(set(entity_ids) - set(cast_entity_ids))
 
-    if cast_entity_ids:
-        await hass.services.async_call(
-            DOMAIN_MP,
-            SERVICE_PLAY_MEDIA,
-            {
-                ATTR_ENTITY_ID: cast_entity_ids,
-                **data,
-                ATTR_MEDIA_EXTRA: {
-                    "stream_type": "LIVE",
-                    "media_info": {
-                        "hlsVideoSegmentFormat": "fmp4",
-                    },
-                },
-            },
-            blocking=True,
-            context=service_call.context,
-        )
-
-    if other_entity_ids:
+    for entities, url_prefix in (
+        (cast_entity_ids, ""),
+        (other_entity_ids, get_url(hass)),
```

Comment:
I'm trying to follow everything going on here, but am missing: How does url signing happen in the case where this passes an absolute url? I realize this is an existing case, so just looking to understand.

Reply:
The stream URL doesn't need authentication. The only reason I trigger the relative URL is because Cast needs to set some extra parameters. It used to be in this method but I have moved it to the play media method as that's where media source would need it applied.

Reply:
Ah right, so the stream url has its own tokens, so it doesn't use authentication.